forge-game/src/main/java/forge/game/ability/SpellAbilityEffect.java

```diff
@@ -621,7 +621,7 @@ public void run() {
     protected static boolean addToCombat(Card c, Player controller, SpellAbility sa, String attackingParam, String blockingParam) {
         final Card host = sa.getHostCard();
         final Game game = controller.getGame();
-        if (!game.getPhaseHandler().inCombat()) {
+        if (!c.isCreature() || !game.getPhaseHandler().inCombat()) {
```

Comment:
the creature check only for attacking

for blocking it might be a planeswalker or battle

Reply:
no, this is not for defender stuff like _Portal Mage_

c is always meant as attacker or blocker which means only creature is legal

> 509.7. If a **creature** is put onto the battlefield blocking...

Reply:
hm ok fine